### PR TITLE
Rewards Banner 

### PR DESCRIPTION
**Pull request by dlipeles**
Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave-intl/publishers/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [ ] Tagged reviewers.

Test Plan:

Reviewer Checklist:

Tests
- [ ] Adequate test coverage exists to prevent regressions

Security:
- [ ] No raw SQL -- Always prefer ActiveRecord query helpers ([more info on StackOverflow](https://stackoverflow.com/questions/41410752/rails-5-sql-injection#41452695))
- [ ] XSS is mitigated -- Avoid `html_safe` and `raw`; escape untrusted content from users and 3rd party APIs ([Rails XSS guide](https://brakemanpro.com/2017/09/08/cross-site-scripting-in-rails) and [OWASP XSS Prevention Cheat Sheet](https://www.owasp.org/index.php/XSS_(Cross_Site_Scripting)_Prevention_Cheat_Sheet))

### CODE REVIEW

File: app/controllers/publishers/site_banners_controller.rb [hunk 1]
```diff
@@ -3,11 +3,11 @@ class Publishers::SiteBannersController < ApplicationController
   MAX_IMAGE_SIZE = 700_000
 
   def new
-    @site_banner = current_publisher.site_banner || SiteBanner.new
+    @site_banner = SiteBanner.find_by(publisher_id: current_publisher.id) || SiteBanner.new
```
Comment (yachtcaptain23) [inline app/controllers/publishers/site_banners_controller.rb]: Preferably continue using `@site_banner = current_publisher.site_banner || SiteBanner.new`

File: app/controllers/publishers/site_banners_controller.rb [hunk 2]
```diff
@@ -20,13 +20,21 @@ def create
     head :ok
   end
 
+  def fetch
```
Comment (yachtcaptain23) [inline app/controllers/publishers/site_banners_controller.rb:23]:
```
def fetch
    data = {}
    data[:backgroundImage] = current_publisher.site_banner.read_only_react_property[:backgroundUrl]
    data[:logoImage] = current_publisher.site_banner.read_only_react_property[:logoUrl]
    render(json: data.to_json)
end
```

**Comment by nvonpentz** [comment]
LGTM

File: app/javascript/packs/brave_rewards_banner_container.jsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    else{
+      rewardsBannerContainer = {height:'590px', width:'840px'}
+    }
+
+    let modeButton;
+
+    if (this.state.mode === 'Edit') {
+      modeButton = <div onClick={() => this.setMode('Preview')} className="brave-rewards-banner-control-bar-save-button" id="edit-button" style={controlButton}>Preview banner</div>;
+    } else {
+      modeButton = <div onClick={() => this.setMode('Edit')} className="brave-rewards-banner-control-bar-save-button" id="preview-button" style={controlButton}>Edit banner</div>;
```
Comment (corymcdonald) [inline app/javascript/packs/brave_rewards_banner_container.jsx:80]: Internationalization of javascript?

**Comment by dlipeles** [comment]
Implemented cropping for photos, @yachtcaptain23 this works well on 0.0.0.0 but will draw a CORS on localhost, please give it a try with S3 if possible. It may just work, but if not I think if you added some allow origin code to the backend we'd be fine.

**Comment by yachtcaptain23** [comment]
Merged in https://github.com/yachtcaptain23/publishers/pull/17

**Pull request closed without merging**